app/controllers/provider_interface/users_controller.rb

```diff
@@ -54,5 +66,21 @@ def edit_user_permissions_store
       key = "edit_user_permissions_wizard_store_#{current_provider_user.id}_#{@provider.id}_#{@provider_user.id}"
       WizardStateStores::RedisStore.new(key: key)
     end
+
+    def provider_permissions_params
+      params.require(:provider_interface_edit_user_permissions_wizard).permit(permissions: [])
+    end
+
+    def permissions_rows
+      ProviderPermissions::VALID_PERMISSIONS.map do |permission|
+
+        {
+          key: t("user_permissions.#{permission}.description"),
+          value: @wizard.permissions.include?(permission.to_s) ? "Yes" : "No",
+          action: t("user_permissions.#{permission}.description"),
+          change_path: edit_provider_interface_organisation_settings_organisation_user_path(@provider, @provider_user),
+        }
+      end
+    end
```

Comment:
Could this be moved to a new component that accepts the wizard, provider and provider user which constructs and renders the rows. This will allow more thorough testing. Happy to discuss.